google/resource_compute_instance.go

```diff
@@ -1164,6 +1150,101 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 				}
 			}
 		}
+		d.SetPartial("network_interface")
+	}
+
+	if d.HasChange("attached_disk") {
+		o, n := d.GetChange("attached_disk")
+
+		// Keep track of disks currently in the instance. Because the google_compute_disk resource
+		// can detach disks, it's possible that there are fewer disks currently attached than there
+		// were at the time we ran terraform plan.
+		currDisks := map[string]struct{}{}
+		for _, disk := range instance.Disks {
+			if !disk.Boot {
+				currDisks[disk.DeviceName] = struct{}{}
+			}
+		}
+
+		// Keep track of previous config's disks.
```

Comment:
This isn't the previous config, it's the state. Terraform does not version configs itself, and so cannot tell you what was in the previous config.

Reply:
Fixed.